drivers/sensor/omron_2smpp03/omron_2smpp03.c

```diff
@@ -0,0 +1,154 @@
+	const char *adc_label;
+	/** ADC channel. */
+	u8_t adc_channel;
+	/** Amplifier gain. */
+	s32_t ampl_gain;
+	/** Amplifier Vref (mV). */
+	s32_t ampl_vref;
+};
+
+static inline struct omron_2smpp03_data *to_data(struct device *dev)
+{
+	return dev->driver_data;
+}
+
+static inline const struct omron_2smpp03_config *to_config(struct device *dev)
+{
+	return dev->config->config_info;
+}
+
+static int omron_2smpp03_sample_fetch(struct device *dev,
+				      enum sensor_channel chan)
+{
+	struct omron_2smpp03_data *data = to_data(dev);
+
+	return adc_read(data->adc, &data->seq);
+}
+
+static int omron_2smpp03_channel_get(struct device *dev,
+				     enum sensor_channel chan,
+				     struct sensor_value *val)
+{
+	struct omron_2smpp03_data *data = to_data(dev);
+	const struct omron_2smpp03_config *cfg = to_config(dev);
+
+	s32_t ref_mv, val_mv;
+	double val_kpa;
+
+	if (chan != SENSOR_CHAN_GAUGE_PRESS) {
+		return -ENOTSUP;
+	}
+
+	ref_mv = (s32_t)adc_ref_internal(data->adc);
+	val_mv = (s32_t)data->raw;
+	adc_raw_to_millivolts(ref_mv, ADC_GAIN, ADC_RES, &val_mv);
+
+	val_kpa = SENSOR_GAIN * (((val_mv - cfg->ampl_vref) / cfg->ampl_gain) -
+				 SENSOR_OFFSET);
+
+	val->val1 = (s32_t)val_kpa;
+	val->val2 = ((s32_t)(val_kpa * 1000000)) % 1000000;
+
+	return 0;
+}
+
+static const struct sensor_driver_api omron_2smpp03_api = {
+	.sample_fetch = &omron_2smpp03_sample_fetch,
+	.channel_get = &omron_2smpp03_channel_get,
+};
+
+static int omron_2smpp03_init(struct device *dev)
+{
+	struct omron_2smpp03_data *data = to_data(dev);
+	const struct omron_2smpp03_config *cfg = to_config(dev);
+
+	int r;
+	struct adc_channel_cfg adc_cfg = { 0 };
+
+	data->adc = device_get_binding(cfg->adc_label);
+	if (!data->adc) {
+		LOG_ERR("Could not obtain ADC device");
+		return -ENODEV;
+	}
+
+	adc_cfg.gain = ADC_GAIN;
+	adc_cfg.reference = ADC_REF;
+	adc_cfg.acquisition_time = ADC_ACQT;
+	adc_cfg.channel_id = cfg->adc_channel;
+
+	r = adc_channel_setup(data->adc, &adc_cfg);
+	if (r < 0) {
+		LOG_ERR("Could not configure ADC cannel (%d)", r);
+		return r;
+	}
+
+	data->seq.buffer = &data->raw;
+	data->seq.buffer_size = sizeof(data->raw);
+	data->seq.channels = BIT(cfg->adc_channel);
+	data->seq.resolution = ADC_RES;
+
+	return 0;
+}
+
+#define OMRON_2SMPP03_INIT(index)					\
+	static struct omron_2smpp03_data omron_2smpp03_data_##index;	\
+									\
+	static const struct omron_2smpp03_config			\
+		omron_2smpp03_cfg_##index = {				\
+		.adc_label = DT_INST_IO_CHANNELS_LABEL(index),		\
+		.adc_channel = DT_INST_IO_CHANNELS_INPUT(index),	\
+		.ampl_gain = DT_INST_PROP(index, ampl_gain),		\
+		.ampl_vref = DT_INST_PROP(index, ampl_vref)		\
+	};								\
+									\
+	DEVICE_AND_API_INIT(omron_2smpp03_##index, DT_INST_LABEL(index),\
+			    &omron_2smpp03_init,			\
+			    &omron_2smpp03_data_##index,		\
+			    &omron_2smpp03_cfg_##index, POST_KERNEL,	\
+			    CONFIG_SENSOR_INIT_PRIORITY,		\
+			    &omron_2smpp03_api)
+
+#define DT_DRV_COMPAT omron_2smpp03
```

Comment:
This is usually defined at the top of the driver implementation file.

Reply:
Is this a requirement? I think it's better to define it here to avoid temptations of using DT macros directly in the driver, so making it non-multiple instance friendly. This forces to collect DT parameters in the instance-specific config structure.

Reply:
No, as far as I know it is not a requirement. But it makes sense to do it consistently across the tree.